src/ExtendedTableWidget.cpp

```diff
@@ -28,59 +29,153 @@ ExtendedTableWidget::ExtendedTableWidget(QWidget* parent) :
 
 void ExtendedTableWidget::copy()
 {
-    // Get list of selected items
-    QItemSelectionModel* selection = selectionModel();
-    QModelIndexList indices = selection->selectedIndexes();
+    QModelIndexList indices = selectionModel()->selectedIndexes();
 
     // Abort if there's nothing to copy
-    if(indices.size() == 0)
-    {
+    if (indices.isEmpty())
         return;
-    } else if(indices.size() == 1) {
-        qApp->clipboard()->setText(indices.front().data().toString());
+    qSort(indices);
+
+    SqliteTableModel* m = qobject_cast<SqliteTableModel*>(model());
+
+    // If single image cell selected - copy it to clipboard
+    if (indices.size() == 1) {
+        QImage img;
+        if (img.loadFromData(m->data(indices.first(), Qt::EditRole).toByteArray())) {
+            qApp->clipboard()->setImage(img);
+            return;
+        }
+    }
+
+    // Check whether selection is rectangular
+    QItemSelection rect(indices.first(), indices.last());
```

Comment:
If we are not implementing multi-selections copy then you can force rectangular selection.